diff --git a/node_modules/c8/lib/parse-args.js b/node_modules/c8/lib/parse-args.js
index 84ff8e3..a448399 100644
--- a/node_modules/c8/lib/parse-args.js
+++ b/node_modules/c8/lib/parse-args.js
@@ -2,7 +2,7 @@ const defaultExclude = require('@istanbuljs/schema/default-exclude')
 const defaultExtension = require('@istanbuljs/schema/default-extension')
 const findUp = require('find-up')
 const { readFileSync } = require('fs')
-const Yargs = require('yargs/yargs')
+const Yargs = require('yargs')
 const { applyExtends } = require('yargs/helpers')
 const parser = require('yargs-parser')
 const { resolve } = require('path')
